foundationdb/src/database.rs

```diff
@@ -64,37 +65,55 @@ impl Database {
     /// It might retry indefinitely if the transaction is highly contentious. It is recommended to
     /// set `TransactionOption::RetryLimit` or `TransactionOption::SetTimeout` on the transaction
     /// if the task need to be guaranteed to finish.
-    pub fn transact<F, Fut, Item>(&self, f: F) -> Box<Future<Item = Fut::Item, Error = FdbError>>
+    pub fn transact<F, Fut, Item, E>(&self, f: F) -> Box<Future<Item = Fut::Item, Error = E>>
     where
         F: FnMut(Transaction) -> Fut + 'static,
-        Fut: Future<Item = Item, Error = FdbError> + 'static,
+        Fut: Future<Item = Item, Error = E> + 'static,
         Item: 'static,
+        E: From<Error> + Send + Sync + 'static,
```

Comment:
@yjh0502 I think this matches what you wanted.

Reply:
Yes, this is what I wanted and I didn't catch it before. Sorry for noise.